main.tf

```diff
@@ -28,14 +31,15 @@ resource "azurerm_kubernetes_cluster" "this" {
     name                = "agentpool"
     vm_size             = "Standard_D4d_v5"
     enable_auto_scaling = true
-    max_count           = 5
-    max_pods            = 110
-    min_count           = 2
-    # node_count although we agreed on 64 - this has to be a number between min_count and max_count
-    node_count = 5
-    os_sku     = "Ubuntu"
-    # os_disk_size_gb - check the GB size of the disk? TODO: research the default size
-    tags = merge(var.tags, var.agents_tags)
+    # autoscaler profile setting on the old module use the configuration
+    enable_host_encryption = true
+    max_count              = 5
+    max_pods               = 110
+    min_count              = 2
+    node_count             = 5
+    os_sku                 = "Ubuntu"
+    tags                   = merge(var.tags, var.agents_tags)
+    zones                  = module.regions.regions_by_name[var.location == null ? local.resource_group_location : var.location].zones
```

Comment:
Could we simplify the expression to `module.regions.regions_by_name[coalesce(var.location, local.resource_group_location)].zones`?